docs/sphinx/using/dynamics.rst

```diff
@@ -415,6 +415,94 @@ backend target.
         *   - `runge_kutta`
             - 1st-order (Euler method), 2nd-order (Midpoint method), and 4th-order (classical Runge-Kutta method).
 
+Batch simulation
+^^^^^^^^^^^^^^^^^
+
+.. _cudensitymat_batching:
+
+CUDA-Q ``dynamics`` target supports batch simulation, which allows users to run multiple simulations simultaneously.
+This batching capability applies to (1) multiple initial states and/or (2) multiple Hamiltonians.
+
+The batching capability can significantly improve performance when simulating many small identical system dynamics, e.g., parameter sweeping, tomography. 
+
+For example, we can simulate the time evolution of multiple initial states with the same Hamiltonian as follows:
+
+.. tab:: Python
+
+    .. literalinclude:: ../snippets/python/using/backends/dynamics_state_batching.py
+        :language: python
+        :start-after: [Begin State Batching]
+        :end-before: [End State Batching]
+
+.. tab:: C++
+
+    .. literalinclude:: ../snippets/cpp/using/backends/dynamics_state_batching.cpp
+        :language: cpp
+        :start-after: [Begin State Batching]
+        :end-before: [End State Batching]
+
+Similarly, we can also batch simulate the time evolution of multiple Hamiltonians as follows:
```

Comment:
Can we show the output of the code and specifically that we get results for each initial state (unlike the simple evolve case). Same comment for all code in this section - show the output

Reply:
Definitely. I've added a code snippet to demonstrate how the batch result can be unpacked, and a simplified sample of the data.